src/adapter/src/catalog.rs

```diff
@@ -4457,7 +4473,23 @@ impl Catalog {
     /// resolve all references.
     #[tracing::instrument(level = "info", skip_all)]
     async fn load_builtin_types(&mut self) -> Result<(), Error> {
-        let persisted_builtin_ids = self.storage().await.load_system_gids().await?;
+        let persisted_builtin_ids: BTreeMap<_, _> = self
```

Comment:
This code is written twice and maybe gnarly enough it should get its own common fn. Does it make sense to add `load_system_gids` as an implemented fn on the trait?

Reply:
I just refactored it a bit to hopefully make it much less gnarly by introducing some new types.